src/casing.ts

```diff
@@ -55,7 +55,12 @@ function toDelimiterCase<T extends string, D extends string>(
  * Transforms a string to camelCase.
  */
 type CamelCase<T extends string> = Words<T> extends [infer first, ...infer rest]
-  ? Join<[Lowercase<Is<first, string>>, ...CapitalizeAll<Is<rest, string[]>>]>
+  ? Join<
+      [
+        Lowercase<Is<first, string>>,
+        ...CapitalizeAll<LowercaseAll<Is<rest, string[]>>>,
+      ]
+    >
   : T
```

Comment:
All the type functions in this file are mapping to the same logic as the runtime logic.
So to keep that pattern this type should probably be defined as follows:
```ts
type CamelCase<T extends string> = PascalCase<T> extends `${infer first}${infer rest}`
  ? `${Lowercase<first>}${rest}`
  : T
```